frontend/src/components/ui/blockEditor/ViewerFragment.jsx

```diff
@@ -0,0 +1,51 @@
+import { Fragment } from "react";
+import { RadioButton, Button } from "@carbon/react";
+import { ViewerCodeMirror } from "./CodeMirrorComponents";
+import { Edit } from "@carbon/icons-react";
+
+export default function ViewerFragment({
+  code,
+  prompt,
+  currentIndex,
+  selectedIndex,
+  handleGenerate,
+  handleEdit,
+}) {
+  const selected = currentIndex == selectedIndex;
+  return (
+    <Fragment key={currentIndex}>
+      <span className="block-editor-prompt">{prompt}</span>
+      <div>
+        <div className="mb-4 flex items-center">
+          <RadioButton
+            checked={selected}
+            onChange={() => handleGenerate(currentIndex)}
+            labelText={`Select Code #${currentIndex}`}
+          />
+        </div>
+        <div
+          className="relative"
+          style={{
+            border: selected ? "2px solid darkorange" : "none",
+          }}
+        >
+          <ViewerCodeMirror
+            key={currentIndex + "-viewer"}
```

Comment:
The fragment already has a key, is this one also necessary?

Reply:
Probably not